### PR TITLE
[libcroco] switch to upstream make build

### PR DESCRIPTION
**Pull request by Neumann-A**

### CODE REVIEW

**Comment by Neumann-A** [comment]
This one is not that important. So until I figure out how to solve the osx failure I'll draft this (probably will be fixed if vcpkg_configure_make setups the compiler for osx; so wait until then. )

**Comment by FrankXie05** [comment]
@Neumann-A Is work still being done for this PR? :)

**Comment by Neumann-A** [comment]
> @Neumann-A Is work still being done for this PR? :)

consider trying to debug the osx issues instead of pinging me ;) might however depend on #25009

File: ports/libcroco/portfile.cmake [hunk 1]
```diff
@@ -1,6 +1,6 @@
 vcpkg_download_distfile(ARCHIVE
-    URLS "https://download.gnome.org/sources/libcroco/0.6/libcroco-0.6.13.tar.xz"
-    FILENAME "libcroco-0.6.13.tar.xz"
+    URLS "https://download.gnome.org/sources/libcroco/0.6/libcroco-${VERSION}.tar.xz"
```
Comment (dg0yt) [inline ports/libcroco/portfile.cmake:2]:
```suggestion
    URLS "https://download.gnome.org/sources/libcroco/0.6/libcroco-${VERSION}.tar.xz"
         "https://www.mirrorservice.org/sites/ftp.gnome.org/pub/GNOME/sources/libcroco/0.6/libcroco-${VERSION}.tar.xz"
```

File: ports/librsvg/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,21 @@
 cmake_minimum_required(VERSION 3.11)
 project(librsvg C)
 
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(LIBCROCO libcroco-0.6 IMPORTED_TARGET)
+pkg_check_modules(GLIB2 glib-2.0 gobject-2.0 gmodule-2.0 gio-2.0 IMPORTED_TARGET)
+pkg_check_modules(CAIRO cairo IMPORTED_TARGET)
+pkg_check_modules(GDK_PIXBUF gdk-pixbuf-2.0 IMPORTED_TARGET)
+pkg_check_modules(PANGO pango pangocairo IMPORTED_TARGET)
```
Comment (dg0yt) [inline ports/librsvg/CMakeLists.txt:9]: How about doing them all in one `pkg_check_modules`?

File: ports/librsvg/CMakeLists.txt [hunk 3]
```diff
@@ -102,9 +117,22 @@ add_library(${LIBRSVG_TARGET} ${LIBRSVG_SOURCES})
...
     -DRSVG_COMPILATION
     -D_CRT_SECURE_NO_WARNINGS
+    -DSRCDIR="${CMAKE_CURRENT_SOURCE_DIR}" 
+    -DGDK_PIXBUF_ENABLE_BACKEND 
+    -DG_LOG_DOMAIN="libpixbufloader-svg"
     -DSRCDIR=""
     $<$<BOOL:${MINGW}>:HAVE_STRTOK_R>
 )
+
+target_link_libraries(rsvg-2.40 PRIVATE
```
Comment (Neumann-A) [inline ports/librsvg/CMakeLists.txt:127]:
```suggestion
target_link_libraries(${LIBRSVG_TARGET} PRIVATE
```

**Review by FrankXie05 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for devendoring the build system!

**Pull request merged**